x-pack/plugins/stack_alerts/public/alert_types/es_query/expression/search_source_expression.tsx

```diff
@@ -0,0 +1,222 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Fragment, useCallback, useEffect, useState } from 'react';
+import './search_source_expression.scss';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiSpacer,
+  EuiTitle,
+  EuiExpression,
+  EuiText,
+  EuiLoadingSpinner,
+  EuiEmptyPrompt,
+  EuiCallOut,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { Filter, ISearchSource } from '../../../../../../../src/plugins/data/common';
+import { EsQueryAlertParams, SearchType } from '../types';
+import {
+  ForLastExpression,
+  RuleTypeParamsExpressionProps,
+  ThresholdExpression,
+  ValueExpression,
+} from '../../../../../triggers_actions_ui/public';
+import { DEFAULT_VALUES } from '../constants';
+import { ReadOnlyFilterItems } from './read_only_filter_items';
+
+export const SearchSourceExpression = ({
+  ruleParams,
+  setRuleParams,
+  setRuleProperty,
+  data,
+  errors,
+}: RuleTypeParamsExpressionProps<EsQueryAlertParams<SearchType.searchSource>>) => {
+  const {
+    searchConfiguration,
+    thresholdComparator,
+    threshold,
+    timeWindowSize,
+    timeWindowUnit,
+    size,
+  } = ruleParams;
+  const [usedSearchSource, setUsedSearchSource] = useState<ISearchSource | undefined>();
+  const [paramsError, setParamsError] = useState<Error | undefined>();
+
+  const [currentAlertParams, setCurrentAlertParams] = useState<
+    EsQueryAlertParams<SearchType.searchSource>
+  >({
+    searchConfiguration,
+    searchType: SearchType.searchSource,
+    timeWindowSize: timeWindowSize ?? DEFAULT_VALUES.TIME_WINDOW_SIZE,
+    timeWindowUnit: timeWindowUnit ?? DEFAULT_VALUES.TIME_WINDOW_UNIT,
+    threshold: threshold ?? DEFAULT_VALUES.THRESHOLD,
+    thresholdComparator: thresholdComparator ?? DEFAULT_VALUES.THRESHOLD_COMPARATOR,
+    size: size ?? DEFAULT_VALUES.SIZE,
+  });
+
+  const setParam = useCallback(
+    (paramField: string, paramValue: unknown) => {
+      setCurrentAlertParams((currentParams) => ({
+        ...currentParams,
+        [paramField]: paramValue,
+      }));
+      setRuleParams(paramField, paramValue);
+    },
+    [setRuleParams]
+  );
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => setRuleProperty('params', currentAlertParams), []);
+
+  useEffect(() => {
+    async function initSearchSource() {
+      try {
+        const loadedSearchSource = await data.search.searchSource.create(searchConfiguration);
+        setUsedSearchSource(loadedSearchSource);
+      } catch (error) {
+        setParamsError(error);
+      }
+    }
+    if (searchConfiguration) {
+      initSearchSource();
+    }
+  }, [data.search.searchSource, searchConfiguration]);
+
+  if (paramsError) {
+    return (
+      <>
+        <EuiCallOut
+          title={i18n.translate('xpack.stackAlerts.searchThreshold.ui.searchSourceParamsError', {
+            defaultMessage: 'Alert creation params error',
```

Comment:
@kertal WDYT, how could we explain search source creation [**error**](https://github.com/elastic/kibana/blob/377f035f7e9785624d3df939fa72100d85461407/x-pack/plugins/stack_alerts/public/alert_types/es_query/expression/search_source_expression.tsx#L82)?

Reply:
There should be an error message in this case, right? And since this is a unlikely error, I'd suggest using it, so when we get screen or feedback, we would have an idea what happened